datastore-v1-proto-client/src/main/java/com/google/datastore/v1/client/RemoteRpc.java

```diff
@@ -123,8 +128,16 @@ public InputStream call(String methodName, MessageLite request) throws Datastore
   }
 
   @VisibleForTesting
-  void setHeaders(MessageLite request, HttpRequest httpRequest) {
+  void setHeaders(
+      MessageLite request, HttpRequest httpRequest, String projectId, String databaseId) {
     httpRequest.getHeaders().put(API_FORMAT_VERSION_HEADER, API_FORMAT_VERSION);
+    StringBuilder builder = new StringBuilder("project_id=");
+    builder.append(projectId);
+    if (!Strings.isNullOrEmpty(databaseId)) {
+      builder.append("&database_id=");
+      builder.append(databaseId);
```

Comment:
This will never contain anything that needs to be encoded, right? Database names are (ALPHA / DIGIT / "-" / "." / "_" / "~")?

Reply:
Yes, that's correct. See: https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases/create#query-parameters